sdk/core/src/services/github/mongo.rs

```diff
@@ -0,0 +1,156 @@
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use async_trait::async_trait;
+
+use serde::{
+    Deserialize,
+    Serialize
+};
+use uuid::Uuid;
+
+use platform::errors::{
+    Error as PlatformError
+};
+
+use platform::persistence::mongodb::{
+    Service as MongoService,
+    MongoDocument,
+    Store as MongoStore,
+    mongo_doc,
+    Context
+};
+use crate::services::github::error::Error;
+
+#[derive(Clone)]
+/// Database service for the GitHubProvider
+pub struct GitHubProviderMongoService {
+    pub(crate) store: Arc<MongoStore>
+}
+
+impl GitHubProviderMongoService {
+
+    /// Conventional Constructor
+    pub fn new(store: Arc<MongoStore>) -> Self {
+        GitHubProviderMongoService {
+            store
+        }
+    }
+
+    /// Given the Mongo Collection we want to put a document into
+    /// and the Harbor Entities, put this document in Mongo
+    pub(crate) async fn create_document(
+        &self, url: String, last_hash: String
+    ) -> Result<GitHubSbomProviderEntry, Error> {
+
+        let mut document = GitHubSbomProviderEntry {
+            id: url.to_string(),
+            last_hash: last_hash.to_string(),
+        };
+
+        match self.insert(&mut document).await {
+            Ok(result) => result,
+            Err(err) => return Err(Error::MongoDb(err)),
+        };
+
+        println!("==> Added NEW Document to MongoDB: {:#?}", document);
+
+        Ok(document)
+    }
+}
+
+#[async_trait]
+impl MongoService<GitHubSbomProviderEntry>
+    for GitHubProviderMongoService {
+
+    fn store(&self) -> Arc<MongoStore> {
+        self.store.clone()
+    }
+
+    /// Insert a document into a [Collection].
+    async fn insert<'a>(
+        &self, doc: & mut GitHubSbomProviderEntry
+    ) -> Result<(), PlatformError> {
+
+        let id = doc.id();
+        if id.is_empty() {
+            doc.set_id(
+                Uuid::new_v4().to_string()
+            );
+        }
+
+        return match self.store().insert::<GitHubSbomProviderEntry>(doc).await {
+            Ok(_rsp) => Ok(()),
+            Err(err) => Err(err)
+        }
+    }
+}
+
+impl Debug for GitHubProviderMongoService {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "GitHubProvider")
+    }
+}
+
+/// Get the default context.  Probably a better one somewhere else
+pub fn get_default_context() -> Context {
+    Context {
+        host: "localhost".to_string(),
+        db_name: "harbor".to_string(),
+        key_name: "id".to_string(),
+        username: "root".to_string(),
+        password: "harbor".to_string(),
+        port: 27017,
+        connection_uri: None,
+    }
+}
+
+/// Struct to define a GitHub Provider document in Mongo
+///
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct GitHubSbomProviderEntry {
```

Comment:
```suggestion
pub struct Commit {
```

And if we add a `timestamp`, this could be the `Commit` type I mentioned earlier, that we store as a vector on the `Repository` entity.

Reply:
We'll do this in a subsequent PR per our conversation.

Reply:
Good suggestion, however watch out for suggesting changes like this because that particular change would break the service because I need to refactor it using the IDE for all instances of `GitHubSbomProviderEntry` to safely become `Commit`s.  So, I actually already had a `Commit` struct in the GitHub Client.  I took that and moved it to the module level, now the mongo portion and the GitHub client are both using it.